.github/workflows/tests.yaml

```diff
@@ -365,6 +365,42 @@ jobs:
         source: /tmp/artifacts
       if: always()
 
+  quota-tests:
+    runs-on: ubuntu-latest
+    needs: build
+    env:
+      S3BACKEND: mem
+      S3QUOTA: scuba
+      QUOTA_ENABLE_INFLIGHTS: "true"
```

Comment:
in-flights are always enabled in tests,  but disabled by default in Cloudserver: so we are introducing risk (aka debt, somehow...), which may hit us once we start using cloudserver with no inflight...

If we do not enable inflight processing by default, this case should be tested as well to avoid future problems. (or we could stick with the inflight enabled for now, even if ZKOP/... force enable the flag, to allow us to later disable it...)

Reply:
We have (unit) test covering when there is no inflight enabled. Not for the functional tests, because the current scuba + mock both use the inflights, I'm not sure this introduces any debt...

If we enable inflights by default, and we someday deploy a real scuba in the CI, we will still have debt when inflights are dropped. If we disable inflights by default, this is the same situation.

What I would suggest is to rename the quota test `quota-tests-with-inflights`, and later we can have another test suite without, if needed... What do you think?

Reply:
(In ZKOP, the flag is enforced anyway)

Reply:
> If we enable inflights by default, and we someday deploy a real scuba in the CI, we will still have debt when inflights are dropped.

In that case, we will actually be making a new change: which must be tested accordingly. In particular, having to disable in-flights when supporting a real scuba is a way to force adding these tests, as otherwise it would not work.

The debt/risk you are avoiding is not about cloudserver itself, but really about having it _silently_ enabled in cloudserver, and thus implicitely enabled... which would indeed break silently if that default value is changed later.

Since the flag is set anyway in ZKOP, we may keep the default in cloudserver to match the test: and be future-proof as Zenko/ZKOP will gracefully handle changes to this default (not very satifying as this is technically a breaking change, but risk should be managed already in this specific case...)

> If we disable inflights by default, this is the same situation.

If we disable inflights by default, but run tests only for another default: we introduce a false sense that the feature is present and tested... And when people start using it, there is a risk that no extra test is added: they will simply build on it, and if the code is good enough not notice the lack of (functional) test...

The default case is the default, so it must be most thoroughly tested. Hence either adding tests for that default case or switching the default.

Reply:
The mock was updated to support the no-inflight mode (S3 API calls will, in this case, directly update the mock, to simulate a system that would work without passing any inflights). At the end, the same tests are running for both scenarios.